Resource cleanup script and documentation added (#111)
magicaks/magicaks: setup.md

* Azure Firewall integrated with network and application rules as recommended by AKS
* Grafana connected to Log Analytics workspace of the cluster is running in Azure Container Instances backed by managed PostgreSQL database

## Resource cleanup

To remove all the resources you can either run the ```terraform destroy``` commands in reverse order, or you can navigate to the ./utils/scripts/ directory and run ```./destroy.sh```. The script will:

* run terraform destroy with auto approve from three folders (3-postprovision, 2-provision-aks, 1-preprovision)
* delete the MagicAKS service identity
* delete MagicAKS service principal

> Sometimes timeing issues may cause the terraform destroy not to complete in time, or fail. Re-running the destroy script will in most cases result in deletion of the resources.

Reviewer: timing

Author: fixed

> Sometimes timeing issues may cause the terraform destroy not to complete in time, or fail. Re-running the destroy script will in most cases result in deletion of the resources.

Reviewer 2: ```suggestion
Sometimes timing issues may cause the terraform destroy not to complete in time, or fail. Re-running the destroy script will in most cases result in deletion of the resources.
```

Reviewer 2: Not sure why this is showing up as outdated now - but its fixed so I'll resolve

> Notice that the destroy script relies on existing resources from which the environment was provisioned (for example .env file, kube.config, etc.). If you are trying to run the terraform destoy from a blank environment, or you don't have access to the original values, you might need to manualy delete the resources. In that case you should:

Reviewer 2: ```suggestion
Notice that the destroy script relies on existing resources from which the environment was provisioned (for example .env file, kube.config, etc.). If you are trying to run the terraform destroy from a blank environment, or you don't have access to the original values, you might need to manually delete the resources. In that case you should:
```

* remove resource groups provisioned by terraform
* remove service principals and groups provisioned by scripts

> * remove any resources that were provisioned manualy (e.g. RBAC groups)

Reviewer 2: ```suggestion
* remove any resources that were provisioned manually (e.g. RBAC groups)
```
